host/src/server/api/v1/proof.rs

```diff
@@ -103,7 +105,9 @@ async fn proof_handler(
         &proof_request.network.to_string(),
     );
 
-    let chain_spec = get_network_spec(proof_request.network);
+    let chain_spec = support_chain_specs
+        .get_chain_spec(&proof_request.network.to_string())
+        .expect("unknown network");
```

Comment:
I would wrap this in a custom error and use the `?` operator so that the client gets a better error